Classes/RedirectService.php

```diff
@@ -107,7 +107,8 @@ protected function buildResponse(ServerRequestInterface $httpRequest, RedirectIn
             $location = $redirect->getTargetUriPath();
 
             if (parse_url($location, PHP_URL_SCHEME) === null) {
-                $location = (string)$httpRequest->getUri()->withQuery('')->withFragment('')->withPath($location);
+                $baseUriForRequest = (string)RequestInformationHelper::generateBaseUri($httpRequest);
+                $location = rtrim($baseUriForRequest, '/') . '/' . ltrim($location, '/');
```

Comment:
Wouldn't be better to assure, that the $baseUriForRequest got from that method always has / has not a trailing slash?

Reply:
You are right, and actually that is the case. It always has a trailing slash

Reply:
`targetUriPath` gets a `ltrim` for slashes in it's constructor.